modules/tinymce/src/plugins/searchreplace/main/ts/ui/Dialog.ts

```diff
@@ -101,47 +99,60 @@ const open = (editor: Editor, currentSearchState: Cell<Actions.SearchState>): vo
     inselection: initialState.inSelection
   };
 
-  const spec: Dialog.DialogSpec<DialogData> = {
+  const getPanelItems = (error: boolean): Dialog.BodyComponentSpec[] => {
+    const items: Dialog.BodyComponentSpec[] = [
+      {
+        type: 'bar',
+        items: [
+          {
+            type: 'input',
+            name: 'findtext',
+            placeholder: 'Find',
+            maximized: true,
+            inputMode: 'search'
+          },
+          {
+            type: 'button',
+            name: 'prev',
+            text: 'Previous',
+            icon: 'action-prev',
+            enabled: false,
+            borderless: true
+          },
+          {
+            type: 'button',
+            name: 'next',
+            text: 'Next',
+            icon: 'action-next',
+            enabled: false,
+            borderless: true
+          }
+        ]
+      },
+      {
+        type: 'input',
+        name: 'replacetext',
+        placeholder: 'Replace with',
+        inputMode: 'search'
+      },
+    ];
+    if (error) {
+      items.push({
+        type: 'alertbanner',
+        level: 'error',
+        text: 'Could not find the specified string.',
+        icon: 'warning',
+      });
+    }
+    return items;
+  };
+
+  const getDialogSpec = (error: boolean, initialData: DialogData): Dialog.DialogSpec<DialogData> => ({
```

Comment:
```suggestion
  const getDialogSpec = (initialData: DialogData, error?: boolean): Dialog.DialogSpec<DialogData> => ({
```

Consider making the `error` variable optional and second in the parameters list. This will allow calling the function without providing a boolean argument first. 

``` ts
getDialogSpec(dialogData)

// instead of
getDialogSpec(false, dialogData)
```

I feel like this is quite more straightforward, unless I am missing some common practice of using `error` as a first argument.

Reply:
Yeah, I know see what you've meant. Maybe it will be a bit more readable to rename the variable to `hasError` to underline the boolean nature of the variable. Object literal is a great idea as well! :)

Reply:
Yeah maybe call it `showNoMatchesAlertBanner` or `showNoMatchesMessage' or something that describes what it does though out the code. Or maybe make it an Optional string with a error message and pass that down the chain not sure what that would be at this point.

Reply:
Done!